src/DRAMPower/DRAMPower/standards/lpddr5/LPDDR5.cpp

```diff
@@ -38,65 +38,140 @@ namespace DRAMPower {
+            BA0, BA1, BA2, BA3, C1, C2, L
+        });
         this->registerPattern<CmdType::RDA>({
-
-                                            });
+            H, H, L, C0, C3, C4, C5,
+            BA0, BA1, BA2, BA3, C1, C2, H
+        });
         this->registerPattern<CmdType::WR>({
-
-                                           });
+            L, H, H, C0, C3, C4, C5,
+            BA0, BA1, BA2, BA3, C1, C2, L
+        });
         this->registerPattern<CmdType::WRA>({
-
-                                            });
+            L, H, H, C0, C3, C4, C5,
+            BA0, BA1, BA2, BA3, C1, C2, H
+        });
         this->registerPattern<CmdType::SREFEN>({
-
-                                               });
+            L, L, L, H, L, H, H,
+            V, V, V, V, V, L, L
+        });
+        this->registerPattern<CmdType::SREFEX>({
+            L, L, L, H, L, H, L,
+            V, V, V, V, V, V, V
+        });
         this->registerPattern<CmdType::PDEA>({
-
-                                             });
+            L, L, L, H, L, H, H,
+            V, V, V, V, V, L, H
+        });
         this->registerPattern<CmdType::PDXA>({
-
-                                             });
+            L, L, L, H, L, H, L,
+            V, V, V, V, V, V, V
+        });
         this->registerPattern<CmdType::PDEP>({
-
-                                             });
+            L, L, L, H, L, H, H,
+            V, V, V, V, V, L, H
+        });
         this->registerPattern<CmdType::PDXP>({
-
-                                             });
+            L, L, L, H, L, H, L,
+            V, V, V, V, V, V, V
+        });
+        this->registerPattern<CmdType::DSMEN>({
+            L, L, L, H, L, H, H,
+            V, V, V, V, V, H, L
+        });
+        this->registerPattern<CmdType::DSMEX>({
+            L, L, L, H, L, H, L,
+            V, V, V, V, V, V, V
+        });
     }
 
     void LPDDR5::handle_interface(const Command &cmd) {
         auto pattern = this->getCommandPattern(cmd);
-        auto length = this->getPattern(cmd.type).size() / commandBus.get_width();
-        this->commandBus.load(cmd.timestamp, pattern, length);
+        auto ca_length = this->getPattern(cmd.type).size() / commandBus.get_width();
+        commandBus.load(cmd.timestamp, pattern, ca_length);
+
+        size_t length = 0;
+
+        switch (cmd.type) {
+            case CmdType::RD:
+            case CmdType::RDA:
+                length = cmd.sz_bits / readBus.get_width();
+                readBus.load(cmd.timestamp, cmd.data, cmd.sz_bits);
+
+                readDQS_c_.start(cmd.timestamp);
+                readDQS_c_.stop(cmd.timestamp + length / this->memSpec.dataRateSpec.dqsBusRate);
+
+                readDQS_t_.start(cmd.timestamp);
+                readDQS_t_.stop(cmd.timestamp + length / this->memSpec.dataRateSpec.dqsBusRate);
+
+                // WCK also during reads
+                if (!memSpec.wckAlwaysOnMode) {
```

Comment:
Where do you add the power for WCK when we are in wckAlwaysOnMode?

Reply:
The power is calculated based on the clock stats in getWindowStats, however there was a bug where the clocks weren't initialized properly. The DQS should be initialized stopped, and the wCK must be initialized depending on the wckAlwaysOnMode mode

Reply:
Ok, and when wckAlwaysOnMode is enabled the last_start is initialized with 0 and we always count the number of cycles between the current time and the start of the simulation, right?

Reply:
yes, correct. that is done in the getWindowStats function